data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconStateHandler.java

```diff
@@ -13,26 +13,62 @@
 
 package tech.pegasys.artemis.beaconrestapi.beaconhandlers;
 
+import static javax.servlet.http.HttpServletResponse.SC_NOT_FOUND;
+
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.apache.tuweni.bytes.Bytes32;
-import tech.pegasys.artemis.beaconrestapi.handlerinterfaces.BeaconRestApiHandler;
+import tech.pegasys.artemis.beaconrestapi.schema.BeaconStateResponse;
+import tech.pegasys.artemis.datastructures.state.BeaconState;
+import tech.pegasys.artemis.provider.JsonProvider;
 import tech.pegasys.artemis.storage.ChainStorageClient;
 
-public class BeaconStateHandler implements BeaconRestApiHandler {
-
+public class BeaconStateHandler implements Handler {
+  public static final String ROUTE = "/beacon/state/";
+  private final Logger LOG = LogManager.getLogger();
   private final ChainStorageClient client;
 
   public BeaconStateHandler(ChainStorageClient client) {
     this.client = client;
   }
 
-  @Override
-  public String getPath() {
-    return "/beacon/state";
+  private BeaconState queryByRootHash(String root) {
+    Bytes32 root32 = Bytes32.fromHexString(root);
+    return client.getStore().getBlockState(root32);
```

Comment:
Pre-genesis `client.getStore()` will return `null` so we need to handle that case.  I suspect you just want to use `client.getBlockState`.

Reply:
ok, i think i've captured that next commit.